third-culture-software/bhima
# Shipment database structure

This PR:
- Adds tables for shipment
- Write into shipment during a stock exit
- Update shipment status during stock entry
- Fix stock entry receipt
- Adds shipment in transit (or partial) in the inventory registry

Closes https://github.com/IMA-WorldHealth/bhima/issues/6351

![image](https://user-images.githubusercontent.com/5445251/153579061-c7b590a4-1896-4abf-a1bc-20fc31268df1.png)

How to test:
- Do stock exit and check inventory registry
- Do stock entry and check inventory registry
- Check the stock entry receipt

UPDATE 2022/02/18

We have added :
- Shipment menu
- Shipment registry
- Shipment creation page

The shipment menu in the navigation tree:
![image](https://user-images.githubusercontent.com/5445251/154673429-5ef3957e-51a2-4f9a-b9a1-dd42a0eec6e0.png)

The shipment registry
![image](https://user-images.githubusercontent.com/5445251/154673509-23563a9d-f72e-4574-9cfd-74b1e91d5446.png)

Actions for a shipment:
![image](https://user-images.githubusercontent.com/5445251/154673847-a4a63961-b737-4d15-a062-86a558ce4b07.png)

Creation page for shipment:
![image](https://user-images.githubusercontent.com/5445251/154673800-7e896173-f3a2-4775-8e8f-2e5e46eaf707.png)

An overview receipt for a shipment:
![image](https://user-images.githubusercontent.com/5445251/154673976-6f28a99f-f37f-4683-88c8-07158a51f1ae.png)

A report for shipments registry
![image](https://user-images.githubusercontent.com/5445251/154674112-fbe5db1c-249f-4bc3-948a-c98048726873.png)

ADDITIONS:
- We have added the table `location` which contains all `depots` by default and we can add other location inside without touching the depot table. This table need a CRUD module

HOW TO TEST:
- Build the database
- Create multiple shipments
- Explore the shipment registry

NOTE: DO NOT MERGE, THIS IS A TRIAL FEEL FREE TO GIVE SUGGESTIONS
